### PR TITLE
Feature form

### PR DESCRIPTION
**Pull request by simonseyock**
read write values to feature properties.
support different forms for different geometry types.
required fields.
ui stuff.

If a field is required, a `(*)` will be shown after the label. If the property `highlightInvalid` of `anol-feature-form` is set to true, the input will be highlighted if it is required and has no value. This is done so we can later set this property if we want to highlight required fields after a user wants to submit invalid data.

### CODE REVIEW

File: src/modules/featureform/featureform-directive.js [hunk 1]
```diff
@@ -21,31 +21,31 @@ angular.module('anol.featureform')
...
                     'feature': '=',
                     'layer': '=', // TODO: is this needed?
-                    'selects': '=', // TODO: is this needed?
-                    'formFields': '=',
-                    'formValues': '=',
-                    'translationNamespace': '@' // TODO: is this needed?
+                    'pointFields': '=?',
+                    'lineFields': '=?',
+                    'polygonFields': '=?',
+                    'highlightInvalid': '=?'
```
Comment (jansule) [inline src/modules/featureform/featureform-directive.js:48]: We can probably set the type of `highlightInvalid` to `'@?'` or `'<?'`, as we do not need a two-way data-binding here. See also https://code.angularjs.org/1.7.9/docs/api/ng/service/$compile#-scope-

File: src/modules/featureform/featureform-directive.js [hunk 2]
```diff
@@ -61,76 +61,78 @@ angular.module('anol.featureform')
...
+                        return angular.isObject(option) ? option.label || option.value : option;
+                    };
+
+                    /**
+                     * @param {FormField} field
+                     * @return {boolean}
+                     */
+                    scope.isValid = function (field) {
+                        if (field.required) {
+                            return !!scope.properties[field.name];
```
Comment (jansule) [inline src/modules/featureform/featureform-directive.js:130]: Does this make a form invalid, if a number field contains a `0`?

File: src/modules/featureform/templates/featureform.html [hunk 3]
```diff
@@ -1,11 +1,35 @@
...
+               class="form-control"
+               type="{{field.type}}"
+               ng-model="properties[field.name]"
+        />
+        <select ng-if="field.type === 'select'"
+                id="input-{{field.name}}"
+                class="form-control"
+                ng-model="properties[field.name]"
+        >
+            <option value="">
```
Comment (jansule) [inline src/modules/featureform/templates/featureform.html:20]: Should we remove this option, if a field is required?

File: src/modules/featureform/featureform-directive.js (hunk 2 above)
Comment (simonseyock) [reply src/modules/featureform/featureform-directive.js:130]: good point

File: src/modules/featureform/templates/featureform.html (hunk 3 above)
Comment (simonseyock) [reply src/modules/featureform/templates/featureform.html:20]: no, then a user might oversee this field and just choose the default value.

**Pull request merged**